exporter/main.py

```diff
@@ -301,17 +309,43 @@ def match(course):
 
     return [course for course in courses if match(course)]
 
+
 def get_all_courses(**kwargs):
     log.info('Retrieving all courses')
-
+    # These are options that don't start with the keyword django
+    other_options = ('lms_config', 'studio_config', 'time_constraint')
     # make a set of fixed arguments, so we can memoize
     kwargs = {
         k: v for k, v in kwargs.iteritems()
-        if k.startswith('django') or k == 'lms_config' or k == 'studio_config'
+        if k.startswith('django') or k in other_options
     }
     kwargs['dry_run'] = False  # always query for course names
     kwargs['limit'] = False  # don't limit number of courses
 
+    # Extract the time constraint option and calculate the end date that
+    # should be used to limit the list of returned course IDs.
+    # Please note that the default time constraint is 3 years
+    constraint = str(kwargs.get('time_constraint', '3'))
+    try:
+        constraint = int(constraint)
+    except ValueError as e:
+        # If the given configuration value cannot be parsed into an integer,
+        # then we quit.
+        error_msg = (
+            'The given time constraint value {c} is not a valid integer. '
+            'The default value of 3 will be used instead: {e}'
+        )
+        raise ExporterFatalError(error_msg.format(c=constraint, e=e))
```

Comment:
Nit: use python3.6+ style f-strings, like
```suggestion
        raise ExporterFatalError(f'The given time constraint value {constraint} is not a valid integer.')
```

Reply:
for simplicity, it's completely acceptable to just raise the stock ValueError class, no need to add complexity by creating a new exception.  This isn't a library or part of a huge codebase, so it's better to keep it simple.
```suggestion
        raise ValueError(f'The given time constraint value {constraint} is not a valid integer.')
```